src/utils/read-file.ts

```diff
@@ -4,15 +4,7 @@ import path from 'path'
 import { findConfigFiles, findFiles } from './find-files'
 import { getFileFromPath } from './get-file-from-path'
 
-export enum FileExtension {
-  JSON = 'json',
-  YAML = 'yaml',
-  YML = 'yml',
-}
-export type File = {
-  contents: EncryptedConfig | Schema
-  filePath: string
-}
+import { File, FileExtension } from '../types'
```

Comment:
are these types used anywhere else or just here?

always a bit wary of throwing too much into these `types.{js|ts}` kind of files as they can grow into black holes over time in my experience

Reply:
Files is used twice, FileExtension once.

I'm also not sure about this. Before going the `types.ts` way, I did some research to check how other TS projects are doing this. But it seemed to me that everyone is doing freestyle (or I don't understand their indivdual requirements well enough).

When `types.ts` becomes too cramped, we can do

```
types/
    io.ts
    config.ts
    ...
```

but same here -- lacking experience to make an educated decision for now.

Reply:
what's the reason we can't handle it the same way as we're doing with flow currently where we define the types in the same file as the actual code? e.g. in flow we would do:

```js
// read-file.js

type FileExtension = 'json' | 'yml'

function something(extension: FileExtension) {
  somelogic()
}
```

Reply:
Works too! However, I use the type to build up an array of file extensions. To do this, I found that's it's much easier/more convenient to enumerate an enum instead of a literal union type:

- literal union type to array: https://stackoverflow.com/questions/44480644/typescript-string-union-to-string-array

- enum to array: `Object.keys(FileExtension)`. Or with named enums: `Object.values(FileExtension`. See `src/read-file.ts` L10.

Reply:
Ok, then i'd suggest the following:

- For widely shared general types, we gather them in some common place (e.g. `types.js`.)
- For types that are very tightly coupled to a specific file, we define them inside that file and import/export them from there

What we should avoid IMO is to re-create our file structure under `src/` in `types/`, e.g. avoid this:

```
- src
   - config.ts
   - read-file.ts
   - some-helper.ts
   - validate.ts
- types
   - config.ts
   - read-file.ts
   - some-helper.ts
   - validate.ts
```

Reply:
Alright. I put `File` and `FileExtension` back into `read-file.ts`. The other types i'd leave in `types.ts` for now as these are rather global + often imported ones.